slowmovie.py

```diff
@@ -10,205 +10,349 @@
-frameCount = int(ffmpeg.probe(inputVid)['streams'][0]['nb_frames'])
-print("there are %d frames in this video" %frameCount)
-
-while 1: 
-
-    if args.random:
-        frame = random.randint(0,frameCount)
-    else: 
-        frame = currentPosition
-
-    msTimecode = "%dms"%(frame*41.666666)
-        
-    # Use ffmpeg to extract a frame from the movie, crop it, letterbox it and save it as grab.jpg 
-    generate_frame(inputVid, 'grab.jpg', msTimecode, width, height)
-    
-    # Open grab.jpg in PIL  
-    pil_im = Image.open("grab.jpg")
-    
-    # Dither the image into a 1 bit bitmap (Just zeros and ones)
-    pil_im = pil_im.convert(mode='1',dither=Image.FLOYDSTEINBERG)
+# ...then try a random video, if --random-file was selected...
+if not currentVideo and args.random_file:
+    videos = list(filter(is_vid, os.listdir(viddir)))
+    if videos:
+        currentVideo = random.choice(videos)
 
-    # display the image 
-    epd.display(epd.getbuffer(pil_im))
-    print('Diplaying frame %d of %s' %(frame,currentVideo))
-    
-    currentPosition = currentPosition + increment 
-    if currentPosition >= frameCount:
-        currentPosition = 0
-        log = open(logdir + '%s<progress'%currentVideo, 'w')
-        log.write(str(currentPosition))
-        log.close() 
-    
-        thisVideo = movieList.index(currentVideo)
-        if thisVideo < len(movieList)-1:
-            currentVideo = movieList[thisVideo+1]
+# ...then try the nowPlaying file, which stores the currently-playing video...
+if not currentVideo and os.path.isfile('nowPlaying'):
+    with open('nowPlaying') as file:
+        lastVideo = file.readline().strip()
+        if os.path.isfile(lastVideo):
+            currentVideo = lastVideo
         else:
-            currentVideo = movieList[0]
+            os.remove('nowPlaying')
+
+# ...then just pick the first video in the videos directory...
+if not currentVideo:
+    # Scan through video directory until you find a video file
+    videos = os.listdir(viddir)
+    for file in videos:
+        if is_vid(file):
+            currentVideo = file
+            break
+
+# ...if none of those worked, exit.
+if not currentVideo:
+    print('No videos found in video directory')
+    sys.exit()
+
+with open('nowPlaying', 'w') as file:
+    file.write(currentVideo)
+
+# Keep track of what video we're playing and which ones are available
+if not args.loop:
+    videos = list(filter(is_vid, os.listdir(viddir)))
+    fileIndex = videos.index(currentVideo)
+
+print(f'Update interval: {args.delay}')
+if not args.random_frames:
+    print(f'Frame increment = {args.increment}')
+
+print(f'With these settings, each minute of 24fps video would take {estimate_runtime(args.delay, args.increment, 60, 24, "d")} to play.')
+print(f'A 120-min movie would last {estimate_runtime(args.delay, args.increment, 120*60, 24, "d")}.')
+
+# Make sure video file passed into CLI is in the videos directory
+if args.file:
+    if args.file in os.listdir(viddir):
```

Comment:
Should this check be done sooner? Might be a hard stop here or at least pull in a random file or something.

Reply:
In my fork I've removed this altogether, allowing for files from any folder.

Reply:
The way I have it working is that --file takes precedence over --dir. The video directory gets set to that file's directory, then plays through starting from there.

Reply:
Nice, that's an elegant solution to that problem. That would still make the options mutually exclusive of each other though right? Passing in ```--file`` and a ```--dir``` option really just ignores the second one. From a usability standpoint forcing one or the other might make more sense for data verification. 

Probably beyond the scope of changes for this PR specifically. Could be separate PR to add that functionality later. The ordering ordering does work in the current code, just some odd edge cases.

Reply:
I think it needs to be there without major changes, because we navigate to the file by appending the Videos directory path with `currentVideo` which is a filename. In missionfloyd's fork there's more complicated path stuff going on. I figure on this we can stay relatively true to the original code without compromising too much—it's not too hard to move files into the Videos directory.

(BTW if you're wondering why I don't check this in an argparser typechecking function, it's because I read in their docs that it's better to have most of that kind of logic outside of the parser.) Ok if I resolve this, while noting it for further development?

Reply:
A bit more: You're right this can cause a hard stop. Made it raise a file not found exception in 04fd8c9